neo/Ledger/Blockchain.cs

```diff
@@ -362,6 +362,7 @@ private RelayResultReason OnNewTransaction(Transaction transaction)
                 UInt256[] delete = mem_pool.Values.AsParallel()
                     .OrderBy(p => p.NetworkFee / p.Size)
                     .ThenBy(p => p.NetworkFee)
+                    .ThenBy(p => p.ReceptionTimestamp)
```

Comment:
You should order it this way also in the code that runs after persisting a block and re-adding transactions to the mem_pool

Reply:
The code may not finish adding all the transactions into the mem_pool that it adds back to message queue by the time of the next consensus. It is therefore necessary to add them into the queue in the right order or they may be processed out of the order desired.

Reply:
In other words: I think you are assuming that the consensus nodes actually have time to process and put all the messages form their queue back in the mem_pool by the time of the next block — I don’t think this assumption is correct.